client/src/main/com/sinch/sdk/domains/sms/GroupsService.java

```diff
@@ -0,0 +1,122 @@
+package com.sinch.sdk.domains.sms;
+
+import com.sinch.sdk.core.exceptions.ApiException;
+import com.sinch.sdk.domains.sms.models.Group;
+import com.sinch.sdk.domains.sms.models.requests.GroupCreateRequestParameters;
+import com.sinch.sdk.domains.sms.models.requests.GroupReplaceRequestParameters;
+import com.sinch.sdk.domains.sms.models.requests.GroupUpdateRequestParameters;
+import com.sinch.sdk.domains.sms.models.requests.GroupsListRequestParameters;
+import com.sinch.sdk.domains.sms.models.responses.GroupsListResponse;
+import java.util.Collection;
+
+/**
+ * Groups Service
+ *
+ * <p>A group is a set of phone numbers (or MSISDNs) that can be used as a target when sending an
+ * SMS. An phone number (MSISDN) can only occur once in a group and any attempts to add a duplicate
+ * are ignored but not rejected.
+ *
+ * @see <a
+ *     href="https://developers.sinch.com/docs/sms/api-reference/sms/tag/Groups">https://developers.sinch.com/docs/sms/api-reference/sms/tag/Groups</a>
+ * @since 1.0
+ */
+public interface GroupsService {
+
+  /**
+   * Retrieve a group
+   *
+   * <p>This operation retrieves a specific group with the provided group ID.
+   *
+   * @param groupId The inbound ID found when listing inbound messages
+   * @return Group associated to groupId
+   * @see <a
+   *     href="https://developers.sinch.com/docs/sms/api-reference/sms/tag/Groups/#tag/Groups/operation/RetrieveGroup">https://developers.sinch.com/docs/sms/api-reference/sms/tag/Groups/#tag/Groups/operation/RetrieveGroup</a>
+   * @since 1.0
+   */
+  Group get(String groupId) throws ApiException;
+
+  /**
+   * Create a group
+   *
+   * <p>A group is a set of phone numbers (MSISDNs) that can be used as a target in the <code>
+   * send_batch_msg</code> operation. An MSISDN can only occur once in a group and any attempts to
+   * add a duplicate would be ignored but not rejected.
+   *
+   * @param parameters Parameters t be used to define group onto creation
+   * @return Created group
+   * @see <a
+   *     href="https://developers.sinch.com/docs/sms/api-reference/sms/tag/Groups/#tag/Groups/operation/CreateGroup">https://developers.sinch.com/docs/sms/api-reference/sms/tag/Groups/#tag/Groups/operation/CreateGroup</a>
+   * @since 1.0
+   */
+  Group create(GroupCreateRequestParameters parameters) throws ApiException;
+
+  Group create() throws ApiException;
```

Comment:
Why two `create`?

Reply:
All parameters are optional; so instead of having `create(null)` calls, simplified by a `create()` function